search/searcher/search_contains_test.go

```diff
@@ -0,0 +1,1040 @@
+//  Copyright (c) 2022 Couchbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package searcher
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/blevesearch/bleve/v2/document"
+)
+
+var (
+	leftRectEdgeMultiPoint [][]float64   = [][]float64{{-1, 0.2}, {-0.9, 0.1}}
+	leftRectWithHole       [][][]float64 = [][][]float64{{{-1, 0}, {0, 0}, {0, 1}, {-1, 1}, {-1, 0}},
+		{{-0.75, 0.25}, {-0.75, -0.75}, {-0.25, 0.75}, {-0.25, 0.25}, {-0.74, 0.25}}}
+	leftRectEdgePoint  []float64   = []float64{-1, 0.2}
+	leftRectMultiPoint [][]float64 = [][]float64{{0.5, 0.5}, {-0.9, 0.1}}
+)
+
+func TestPointPolygonContains(t *testing.T) {
+	tests := []struct {
+		QueryShape       []float64
+		DocShapeVertices [][][]float64
+		DocShapeName     string
+		Expected         []string
+		Desc             string
+		QueryType        string
+	}{
+		{
+			QueryShape:       rightRectPoint,
+			DocShapeVertices: rightRect,
+			DocShapeName:     "polygon1",
```

Comment:
DocShapeName looks un used and contains an invalid value?

Reply:
I use it here - https://github.com/metonymic-smokey/bleve/blob/dd5928c00253f2a0c9fd50e1c88cbe2e87b974c1/search/searcher/search_contains_test.go#L61 
Why is the value invalid?

Reply:
my bad.. I mistaken that for the shape type that was getting parameterised later to the runXXX functions.. My thinking was that all test specific values would be driven by the table and the rest would be a few utility functions that are invoked with those test entries from the tables.